docs/rollen.md

```diff
@@ -0,0 +1,263 @@
+# Betrokkenen/rollen with authentication context
+
+Example API calls with required fields and dummy data, the focus is on the shape of
+`authenticatieContext`.
+
+## DigiD / Without mandate
+
+### DigiD - initiator
+
+```http
+POST /zaken/api/v1/rollen
+Content-Type: application/json
+
+{
+  "zaak": "http://example.com",
+  "betrokkeneType": "natuurlijk_persoon",
+  "roltype": "http://example.com/roltype-initiator",
+  "roltoelichting": "Created zaak",
+  "betrokkeneIdentificatie": {
+    "inpBsn": "123456782"
+  },
+  "authenticatieContext": {
+    "source": "digid",
+    "levelOfAssurance": "urn:oasis:names:tc:SAML:2.0:ac:classes:MobileTwoFactorContract"
+  }
+}
+```
+
+### DigiD - belanghebbende
+
+```http
+POST /zaken/api/v1/rollen
+Content-Type: application/json
+
+{
+  "zaak": "http://example.com",
+  "betrokkeneType": "natuurlijk_persoon",
+  "roltype": "http://example.com/roltype-belanghebbende",
+  "roltoelichting": "Voogd",
+  "betrokkeneIdentificatie": {
+    "inpBsn": "123456782"
+  },
+  "authenticatieContext": null
+}
+```
+
+## DigiD / With mandate
+
+### DigiD - initiator
+
+The authorizee is the initiator of the case (cleared up from VNG github), the representee
+is a "belanghebbende".
+
+```http
+POST /zaken/api/v1/rollen
+Content-Type: application/json
+
+{
+  "zaak": "http://example.com",
+  "betrokkeneType": "natuurlijk_persoon",
+  "roltype": "http://example.com/roltype-initiator",
+  "roltoelichting": "Created zaak",
+  "indicatieMachtiging": "gemachtigde",
+  "betrokkeneIdentificatie": {
+    "inpBsn": "123456782"
+  },
+  "authenticatieContext": {
+    "source": "digid",
+    "levelOfAssurance": "urn:oasis:names:tc:SAML:2.0:ac:classes:MobileTwoFactorContract"
+    "representee": {
+      "identifierType": "bsn",
+      "identifier": "111222333"
+    },
+    "mandate": {
+      "services": [
+        {"id": "5628edbd-333e-460d-8a69-8f083b8cf1b8"}
+      ]
+    }
+  }
+}
+```
+
+### DigiD - belanghebbende
+
+```http
+POST /zaken/api/v1/rollen
+Content-Type: application/json
+
+{
+  "zaak": "http://example.com",
+  "betrokkeneType": "natuurlijk_persoon",
+  "roltype": "http://example.com/roltype-belanghebbende",
+  "roltoelichting": "Voogd",
+  "indicatieMachtiging": "machtiginggever",
+  "betrokkeneIdentificatie": {
+    "inpBsn": "111222333"
+  },
+  "authenticatieContext": null
+}
+```
+
+## DigiD // Query patterns
+
+1. Show me my cases:
+
+    - show zaken WHERE -- cases opened for myself
+        betrokkeneIdentificatie = :ownBsn
+        AND rol.omschrijvingGeneriek = initiator
```

Comment:
there may be multiple authorizees and authorizers (in theory), so yes, it's needed

additionally, for older data from before `indicatieMachtiging` was added, or rollen being created without this attribute set, the `initiator` rolOmschrijvingGeneriek is the only guarantee you have

Reply:
I've added the option to filter on `rol.omschrijvingGeneriek` to the docs, but since we don't enforce it with validation I don't include it to `rol__machtiging` filter